helper/validation/path.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package validation
+
+import (
+	"github.com/hashicorp/go-cty/cty"
+)
+
+// PathEquals compares two Paths for equality. For cty.IndexStep,
+// unknown key values are treated as an Any qualifier and will
+// match any index step of the same type.
+func PathEquals(p cty.Path, other cty.Path) bool {
```

Comment:
nit: Just a naming thing, since this is kind of fuzzy/expression matching, maybe something like `PathMatches`? Perhaps we can do a formal bikeshed for naming later 😄